src/DicomWebClient/API/IQidoService.cs

```diff
@@ -0,0 +1,76 @@
+        /// </summary>
+        /// <param name="queryParameters">A dictionary object where the <c>Key</c> contains the DICOM tag
+        /// or keyword of an attribute and the <c>Value</c> contains the expected value to match.</param>
+        /// <param name="fieldsToInclude">Liist of DICOM tags of name of the DICOM tag to be included in the response.</param>
+        IAsyncEnumerable<string> SearchForStudies(Dictionary<string, string> queryParameters, List<string> fieldsToInclude);
+
+        /// <summary>
+        /// Search for studies based on provided query parameters with additional DICOM fields to be included in the response message.
+        /// </summary>
+        /// <param name="queryParameters">A dictionary object where the <c>Key</c> contains the DICOM tag
+        /// or keyword of an attribute and the <c>Value</c> contains the expected value to match.</param>
+        /// <param name="fieldsToInclude">Liist of DICOM tags of name of the DICOM tag to be included in the response.</param>
+        /// <param name="fuzzyMatching">Whether fuzzy semantic matching should be performed.</param>
+        IAsyncEnumerable<string> SearchForStudies(Dictionary<string, string> queryParameters, List<string> fieldsToInclude, bool fuzzyMatching);
+
+        /// <summary>
+        /// Search for studies based on provided query parameters with additional DICOM fields to be included in the response message.
+        /// </summary>
+        /// <param name="queryParameters">A dictionary object where the <c>Key</c> contains the DICOM tag
+        /// or keyword of an attribute and the <c>Value</c> contains the expected value to match.</param>
+        /// <param name="fieldsToInclude">Liist of DICOM tags of name of the DICOM tag to be included in the response.</param>
+        /// <param name="limit">Maximum number of results to be returned.</param>
+        IAsyncEnumerable<string> SearchForStudies(Dictionary<string, string> queryParameters, List<string> fieldsToInclude, bool fuzzyMatching, int limit);
+
+        /// <summary>
+        /// Search for studies based on provided query parameters with additional DICOM fields to be included in the response message.
+        /// </summary>
+        /// <param name="queryParameters">A dictionary object where the <c>Key</c> contains the DICOM tag
+        /// or keyword of an attribute and the <c>Value</c> contains the expected value to match.</param>
+        /// <param name="fieldsToInclude">Liist of DICOM tags of name of the DICOM tag to be included in the response.</param>
+        /// <param name="limit">Maximum number of results to be returned.</param>
+        /// <param name="offset">Number of results to be skipped.</param>
+        IAsyncEnumerable<string> SearchForStudies(Dictionary<string, string> queryParameters, List<string> fieldsToInclude, bool fuzzyMatching, int limit, int offset);
```

Comment:
recommend `IReadOnlyDictionary<K, V>` and `IReadOnlyList<T>` be used

Reply:
Fixed 🙇